geth/jail/cell.go

```diff
@@ -86,11 +88,39 @@ func (c *Cell) Stop() error {
 // event queue loop and schedules for immediate execution.
 // Intended to be used by any cell user that want's to run
 // async call, like callback.
-func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) {
+func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) error {
 	task := looptask.NewCallTask(fn, args...)
-	// Add a task to the queue.
-	c.loop.Add(task)
-	// And run the task immediately.
-	// It's a blocking operation.
-	c.loop.Ready(task)
+	errChan := make(chan error, 1)
+	defer close(errChan)
+
+	go func() {
+		err := c.loop.Add(task)
```

Comment:
As we're free to change `loop` package as we want, I would add here a new method for "adding the task for immediate execution`. That would reduce the number of error checking and will make the code more clear.